### DISCUSSION TITLE
Vim/Evil mode command and Russian language auto select

### DISCUSSION THREAD

**Discussion by ghost355** [syl20bnr/spacemacs#15726] [Q&A] [answered]
Is there an ability for auto selecting (return to English) from Russian when I quit an insert mode and return to the last language when I turn back to the insert mode. It's very useful when I write something in Russian and press Esc for navigation the text. This work well  in vim with https://github.com/lyokha/vim-xkbswitch. I'm looking for this feature in Spacemacs. Or any other way to make my life easy without regular switching language layouts.

**Comment by TheBB** [comment]
There are hooks for entering and exiting each evil state where you can attach a function that does arbitrary things. If you can figure out how to change the language in Emacs Lisp (e.g. by calling an executable), that should solve the problem.

**Comment by ghost355** [comment]
Thanks. I think this https://github.com/vovkasm/input-source-switcher could help in this task.This small utility for Apple OS X allows to easily switch input sources from a command line.

**Comment by TheBB** [comment]
So you would want something along these lines.

``` lisp
(add-hook 'evil-insert-state-entry-hook
          (lambda ()
            (shell-command "issw russian")))
(add-hook 'evil-insert-state-exit-hook
          (lambda ()
            (shell-command "issw english")))
```

**Comment by ghost355** [comment]
With exit mode looking OK, but when I enter in Insert mode I have to switch to the last layouts English or Russian which was active before exit from Insert mode ))).

**Comment by ghost355** [comment]
`issw` can return the active input source. When I exit from Insert mode I have to recieve actual layout and save it. Then switch to English.  And when I return to Insert I use `issw` with saved state

**Comment by TheBB** [comment]
Okay, then make a variable and use it to store the info. You can grab the output of a shell command with `shell-command-to-string`.

**Comment by ghost355** [comment]
Thanks, I'll need some times to learn ELisp :)

**Comment by TheBB** [comment]
Since this is not exactly the place, I am closing this now.

You can pop on the gitter if you like.

**Comment by ghost355** [comment]
Thanks. You show me the way

**Comment by kai11** [comment]
Working solution for Ubuntu in https://gist.github.com/kai11/bb5c2fa03cb21884bfcb76c01d1632c9

**Comment by ghost355** [comment]
Thanks. I already did it for Mac )))
https://gist.github.com/ghost355/a967417bb7e826e833d4

**Comment by malashinroman** [comment, marked as answer]
> Working solution for Ubuntu in https://gist.github.com/kai11/bb5c2fa03cb21884bfcb76c01d1632c9

Thanks, I updated your solution to newer version of Ubuntu.

https://github.com/malashinroman/dotfiles/blob/571cb33d92a6a26f0e88ed2b22785274ff1a38ae/.spacemacs#L613-L656
